### PR TITLE
Sidekick fixes and enhancements

### LINKED ISSUE #548: Sidekick: js error in publish action
```
app.js:168 Uncaught (in promise) TypeError: Cannot read property 'notify' of undefined
    at sendPurge (app.js:168)
    at async HTMLButtonElement.action (app.js:192)
```

### LINKED ISSUE #549: Sidekick: extend notify method to support more use cases
For example more than one text block or paragraph, pre-formatted blocks etc.

### LINKED ISSUE #552: Sidekick: doesn't work without branch/ref
The configurator will use `master` as default branch/ref, but the sidekick should ignore that when building the `innerHost` and `outerHost` hostnames. Otherwise users will always have to prefix their URLs with `master--` in order to use the sidekick.

### LINKED ISSUE #554: Sidekick: add margins for multiple button rows

### PR DESCRIPTION
**Pull request by rofe**
Fix #548 
Fix #549 
Fix #552 
Fix #554

### CODE REVIEW

**Comment by rofe** [comment]
(the failed `run_diff_tests` ran through mean while. it was a network timeout unrelated to the changes in this PR)

File: tools/sidekick/app.js [hunk 1]
```diff
@@ -49,7 +49,7 @@
      */
     _initConfig(cfg = {}) {
       const prefix = (cfg && cfg.owner && cfg.repo)
-        ? `${cfg.ref ? `${cfg.ref}--` : ''}${cfg.repo}--${cfg.owner}`
+        ? `${cfg.ref && cfg.ref !== 'master' ? `${cfg.ref}--` : ''}${cfg.repo}--${cfg.owner}`
```
Comment (kptdobe) [inline tools/sidekick/app.js:52]: humm... we are going away from `master`. And you cannot be sure if default is `main` or `master`... Is it really necessary ?

**Review by kptdobe — CHANGES_REQUESTED** [review]

File: tools/sidekick/app.js (hunk 1 above)
Comment (rofe) [reply tools/sidekick/app.js:52]: Well, yes, because in both cases, it would not be part of the URL

Comment (rofe) [reply tools/sidekick/app.js:52]: I'll include `main` for now

Comment (kptdobe) [reply tools/sidekick/app.js]: Ok, cool. But I could be picky saying there is no test for this specific use case ;)

**Review by kptdobe — APPROVED** [review]

**Pull request merged**